### PR TITLE
[3.3] bpo-22928: Disabled HTTP header injections in http.client.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Original patch by Demian Brecht.
(cherry picked from commit a112a8ae47813f75aa8ad27ee8c42a7c2e937d13)

https://bugs.python.org/issue22928

### CODE REVIEW

**Comment by vstinner** [comment]
See my comment on the bpo: http://bugs.python.org/issue22928#msg299053

File: Lib/http/client.py [hunk 1]
```diff
@@ -216,6 +217,34 @@
...
+# tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
+#                / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
+#                / DIGIT / ALPHA
+#                ; any VCHAR, except delimiters
+#
+# VCHAR defined in http://tools.ietf.org/html/rfc5234#appendix-B.1
+
+# the patterns for both name and value are more leniant than RFC
+# definitions to allow for backwards compatibility
+_is_legal_header_name = re.compile(rb'[^:\s][^:\r\n]*\Z').match
```
Comment (vstinner) [inline Lib/http/client.py:245]: Python 2.7 regex starts with \A. Is it useful?

Comment (serhiy-storchaka) [reply Lib/http/client.py:245]: No. `match()` always matches from the start.

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**